cctools-port 949.0.1 broke various builds on macOS

**Describe the bug**
Ruby configure step has been failing on darwin since https://github.com/NixOS/nixpkgs/commit/3c6bd61560d042ce1e366989b92b615dfa1dffb3 with the following error:

```
checking whether CFLAGS is valid... yes
checking whether LDFLAGS is valid... no
configure: error: something wrong with LDFLAGS=""
```

This appears to be due to the new linker warning `ld: warning: passed two min versions (10.12.0, 10.12) for platform macOS`, as adding `LDFLAGS="-Wl,-w"` to ruby allows it to build successfully. 

git bisect log:
<details>

```
git bisect start
# bad: [92c96b799528442f6799f829d11602786689b28e] pkgs-lib/tests/formats: fix whitespace after PR #100372
git bisect bad 92c96b799528442f6799f829d11602786689b28e
# good: [53ce0bfc904a8e0dbe0d81cc1624815d4d8a038f] Merge pull request #95592 from risicle/ris-flink-1.11.1
git bisect good 53ce0bfc904a8e0dbe0d81cc1624815d4d8a038f
# good: [4c6dad5e826929ecd125472b4608b493b3efc24e] lsd: add maintainer
git bisect good 4c6dad5e826929ecd125472b4608b493b3efc24e
# good: [6d174dd5e0650073971a75efec218c79cbce951f] linux: Remove 5.7
git bisect good 6d174dd5e0650073971a75efec218c79cbce951f
# bad: [27143f8e022cbc9e0a480c3b89d757920aef55f6] nodejs-12_x: 12.18.4 -> 12.19.0
git bisect bad 27143f8e022cbc9e0a480c3b89d757920aef55f6
# bad: [0a5be167a3b5808c2b42a8660cf015c79965b204] Merge pull request #97098 from volth/cpan2nix-2020-09-04
git bisect bad 0a5be167a3b5808c2b42a8660cf015c79965b204
# good: [e9f7e3a97350133bdcd699ef44fbbf9d5678968b] [cpan2nix] perlPackages.PPIxRegexp: 0.072 -> 0.074
git bisect good e9f7e3a97350133bdcd699ef44fbbf9d5678968b
# good: [f728bea688227536a73b382ec1a2afb8be2854d0] [cpan2nix] perlPackages.CatalystViewTT: 0.44 -> 0.45
git bisect good f728bea688227536a73b382ec1a2afb8be2854d0
# bad: [5e68d203f3f8abf8f928fb6788487b83e3a0e8ec] opencv2: mark as broken
git bisect bad 5e68d203f3f8abf8f928fb6788487b83e3a0e8ec
# bad: [9112a9d882ed58d1823db8558d0bc8ceb2f33a5e] Merge pull request #93596 from obsidiansystems/ios-13
git bisect bad 9112a9d882ed58d1823db8558d0bc8ceb2f33a5e
# good: [1ca7c6e76a5138f7f0f3661c4d372fe15c0905c2] Merge pull request #97485 from marsam/update-nodejs-14_x
git bisect good 1ca7c6e76a5138f7f0f3661c4d372fe15c0905c2
# bad: [95eabdfd5f6197c83e9e5f53ddb65dbbea3c13fd] xcode/sdk-pkgs.nix: set -platform_version in addition to -miphoneos-version-min
git bisect bad 95eabdfd5f6197c83e9e5f53ddb65dbbea3c13fd
# bad: [3c6bd61560d042ce1e366989b92b615dfa1dffb3] darwin.cctools: 927.0.2 -> 949.0.1
git bisect bad 3c6bd61560d042ce1e366989b92b615dfa1dffb3
# good: [39c536c8db4e0982f89b67a194cd36ba9f4fbdd5] darwin.libtapi: update, fix installTargets
git bisect good 39c536c8db4e0982f89b67a194cd36ba9f4fbdd5
# first bad commit: [3c6bd61560d042ce1e366989b92b615dfa1dffb3] darwin.cctools: 927.0.2 -> 949.0.1
```
</details>

~~Further bisection on the cctools-port repo itself is unproductive, simply leading to the mass update https://github.com/tpoechtrager/cctools-port/commit/466063c~~ See https://github.com/NixOS/nixpkgs/issues/101330#issuecomment-716068007 for the offending code in cctools.

**Hydra build**
https://hydra.nixos.org/build/128219205

**Notify maintainers**
cctools: @matthewbauer 
ruby: @vrthra @manveru @marsam

--- a/pkgs/development/interpreters/ruby/default.nix
+++ b/pkgs/development/interpreters/ruby/default.nix
@@ -125,6 +125,10 @@ let
             "--with-out-ext=tk"
             # on yosemite, "generating encdb.h" will hang for a very long time without this flag
             "--with-setjmp-type=setjmp"
+            # silence linker warnings after upgrading darwin.cctools to 949.0.1,
+            # which ruby treats as problem with LDFLAGS
+            # https://github.com/NixOS/nixpkgs/issues/101330
+            "LDFLAGS=-Wl,-w"
           ]
           ++ op (stdenv.hostPlatform != stdenv.buildPlatform)
              "--with-baseruby=${buildRuby}";

